### PR TITLE
Gutenboarding: Collapse launch button + plugin icons on mobile

### LINKED ISSUE #48337: Launch button and plugin icons need menu collapse on mobile
The block editor navigation bar needs to collapse branding icons and extra buttons on mobile views.

Simple sites are fine in most cases, the problems arise when there are too many dynamic additions to the menu bar.

Examples I found:
- A8c icon (global styles?) + Jetpack icon + Launch button
- Draft save cloud icon
- Screen widths where the undo/redo actions aren't yet collapsed

**Note**: Launch button will soon be added also to the editor header in wp-admin #47803

### PR DESCRIPTION
**Pull request by StefanNieuwenhuis**
#### Changes proposed in this Pull Request

* Tidy up the icons in the header on mobile viewport widths and collapse relevant actions.

#### Testing instructions

* Checkout this branch
* Ensure your site is sandboxed and you're logged in to your sandbox environment
* Navigate to `apps/editing-toolkit` and run `yarn dev --sync`
* Open a post or page on your chosen site to go to the editor
* Shrink your viewport down to mobile size and observe the collapsed menu in the editor

**Before**
Screenshot 2021-01-22 at 10 38 46

**After**
Screenshot 2021-01-22 at 10 38 55

Video: https://www.loom.com/share/2709f1194d5a4e48b2fe154179089051

Fixes #48337

### CODE REVIEW

File: apps/editing-toolkit/package.json [hunk 1]
```diff
@@ -126,7 +129,7 @@
 		"@wordpress/html-entities": "*",
 		"@wordpress/i18n": "*",
 		"@wordpress/icons": "*",
-		"@wordpress/interface": "^0.7.5",
+		"@wordpress/interface": "^1.0.0",
```
Comment (tjcafferkey) [inline apps/editing-toolkit/package.json]: We need to double check this package upgrade wont negatively impact other features that are using it.

File: apps/editing-toolkit/editing-toolkit-plugin/editor-header-buttons/src/index.tsx [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+				className="my-container-class-name"
+				contentClassName="my-popover-content-classname"
+				position="bottom center"
+				renderToggle={ ( { isOpen, onToggle } ) => (
+					<Button isPrimary onClick={ onToggle } aria-expanded={ isOpen }>
+						Update
+					</Button>
+				) }
+				renderContent={ () => (
+					<div className="dropdown-content">
```
Comment (tjcafferkey) [inline apps/editing-toolkit/editing-toolkit-plugin/editor-header-buttons/src/index.tsx:44]: These components may require additional props which we will also need to look into. You can find their usages [here in Gutenberg](https://github.com/WordPress/gutenberg/blob/master/packages/edit-post/src/components/header/index.js#L67-L94)

File: apps/editing-toolkit/package.json (hunk 1 above)
Comment (simison) [reply apps/editing-toolkit/package.json]: I think there recently was packages upgrade so rebasing might sort it.

Comment (Copons) [reply apps/editing-toolkit/package.json]: `@wordpress/interface` was replaced by `@wordpress/edit-site` in the toolkit in #48729, so I'd say that upgrading to 1.0.0 shouldn't cause any regressions.

**Comment by Copons** [comment]
I'm personally very wary of introducing new .com-only UI patterns, especially considering that this issue is as valid on .com as it is on .org — maybe even more, as .org users might be more inclined to install dozens of plugins adding their icons to the header.
If we really want to go ahead with a mobile-only dropdown replacing plugins and other header actions, I think we should propose it on the Gutenberg repo instead (assuming it's not something being already worked on).

I also fear that this approach might prove to be too complicated to maintain, given the amount of variations the editor header can have. What about draft posts? What about the non-post editors?

As a quick stop-gap, we might want to start by simply hiding the plugins icons, which are still accessible through the More menu.

Fixing the header actions (switch to draft, preview, update/publish) is more difficult, as they don't exist anywhere else, and aren't semantically strictly related.
We also plug into the Preview button on .com to use the Calypso preview component. I haven't checked so it might work fine, but it's yet another example of things that could go wrong. 😄 

---

tl;dr: why not just hide the plugin icons for now?

File: apps/editing-toolkit/editing-toolkit-plugin/editor-header-buttons/index.php [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+
+	wp_set_script_translations( 'a8c-fse-editor-header-buttons-script', 'full-site-editing' );
+
+	wp_localize_script(
+		'a8c-fse-editor-header-buttons-script',
+		'wpcomEditorSiteLaunch',
+		array(
+			'locale' => determine_locale(),
+		)
+	);
```
Comment (p-jackson) [inline apps/editing-toolkit/editing-toolkit-plugin/editor-header-buttons/index.php:44]: This looks like it was copy/pasted from another `php` file, but we don't need it here. This code adds a `window.wpcomEditorSiteLaunch.locale` global, but that global is not used by the editor header buttons.

We should delete these lines.

**Comment by p-jackson** [comment]
The iPhone 8 layout works with just the plugin icons hidden

Screenshot 2021-01-27 at 11 39 48 AM

> tl;dr: why not just hide the plugin icons for now?

I agree with this. Hiding the plugin icons fixes the layout issue and doesn't get us into the sticky situation of moving DOM items around. This has been flakey in the past and caused crashes in the past (if we manually update the DOM then React can break if it ever tries to re-render the header). Also sometimes the launch button is shown in the header and we'd need to duplicate all the logic for that too.

**Comment by p-jackson** [comment]
If we do just hide the plugin buttons then I think all this PR needs is this bit:
```scss
@import '~@wordpress/base-styles/mixins';
@import '~@wordpress/base-styles/variables';
@import '~@wordpress/base-styles/breakpoints';

.interface-pinned-items > button:not(:first-child) {
	@media ( max-width: $break-medium ) {
		display: none;
	}
}
```

I'll post about this to make sure happiness are aware of what's coming up.

**Comment by p-jackson** [comment]
I've created another PR that fixes the issue by only hiding the plugin icons #49329 @Copons. There's nothing wrong with this PR (in fact I copy/pasted most of the code 😄) but I felt I was going to delete most of it anyway, and I thought we may still want to be able to easily refer to these changes.

**Comment by p-jackson** [comment]
A temporary fix has been merged in #49329 instead.

There's also a fix for the underlying problem in Wordpress/gutenberg#28521 and Wordpress/gutenberg#28526 that'll be shipped in GB 9.9
